starting movie-app git repo

pblobp opened the pull request.

arctouch-emanuelmachado commented on app/src/main/java/com/arctouch/pablo/movieapp/activity/ActMovies.java:
```diff
@@ -0,0 +1,284 @@
+    private boolean isLoading = false;
+
+    // genres and upcoming movies
+    private List<ResponseGenre> genres = new ArrayList<>();
```
What's this list for? It doesn't seem to be used...

arctouch-emanuelmachado commented on app/src/main/java/com/arctouch/pablo/movieapp/activity/ActMovies.java:
```diff
@@ -0,0 +1,284 @@
+        super.onCreate(savedInstanceState);
+
+        //
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbarMovies);
```
Do we really need the cast to `Toolbar`? Didn't `findViewById` get generified in sdk 26 (which we're using)?

arctouch-emanuelmachado commented on app/src/main/java/com/arctouch/pablo/movieapp/activity/ActMovies.java:
```diff
@@ -0,0 +1,284 @@
+
+                    // add to adapter
+                    adtMovie.addAll(response.body().getResults());
+                    listMovies.setAdapter(adtMovie);
```
When you change the RecyclerView's adapter, you force it to update, which causes it to scroll back to the beginning every time it loads more movies. Do you know what to change so you don't have to change the adapter on every movie update/

arctouch-emanuelmachado commented on app/src/main/java/com/arctouch/pablo/movieapp/adapter/AdtMovie.java:
```diff
@@ -0,0 +1,190 @@
+     * @param mcList
+     */
+    public void addAll(List<ResponseUpcomingMovie> mcList) {
+        for (ResponseUpcomingMovie mc : mcList) {
```
Is there a reason you didn't use `movies.addAll(mcList)`?

arctouch-emanuelmachado commented on app/src/main/java/com/arctouch/pablo/movieapp/activity/ActMovies.java:
```diff
@@ -0,0 +1,284 @@
+                .build();
+
+        // instance api
+        ActMovies.api = retrofit.create(MoviesAPI.class);
```
If you're creating the Retrofit instance here, why do you need the `RESTAdapter` class?

arctouch-emanuelmachado commented on app/src/main/java/com/arctouch/pablo/movieapp/model/ResponseUpcomingMovie.java:
```diff
@@ -0,0 +1,128 @@
+     * @param genres
+     * @return
+     */
+    public String getGenresName(List<ResponseGenre> genres) {
```
Isn't this the same method we have in `AdtMovie`? (Though this one seems incomplete...)

arctouch-emanuelmachado commented on app/src/main/res/layout/content_act_movie.xml:
```diff
@@ -0,0 +1,83 @@
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/textViewDetailReleaseDate" />
+
+</android.support.constraint.ConstraintLayout>
```
This layout is ok, but if we turn the phone sideways (to landscape mode), then we lose half of the information... What do you think we could do avoid that problem?

arctouch-emanuelmachado commented on app/build.gradle:
```diff
@@ -0,0 +1,34 @@
+    compile 'com.squareup.picasso:picasso:2.5.2'
+    implementation 'com.android.support:cardview-v7:26.1.0'
+    compile 'com.squareup.retrofit2:retrofit:2.3.0'
+    compile 'com.squareup.retrofit2:converter-gson:2.3.0'
```
Why do you have both `implementation` and `compile` dependencies here? Shouldn't they all be `implementation`?

pblobp commented:
Hi Bill, I’m fine, thank you, and you?

I’m happy for your review.
Lets go, I will answer between your below questions, ok?

Att.

Pablo Boff Pigozzo
(48) 99186-0790 / pblobp@gmail.com

On 28 Nov 2017 08:50 -0200, Emanuel Machado da Silva <notifications@github.com>, wrote:
> @arctouch-emanuelmachado commented on this pull request.
> Hi Pablo, my name is Bill and I work for ArcTouch. How's it going?
> First off, I'd like to thank you for taking the time to go through this small exercise with us, we appreciate it. I've reviewed your code and added a few questions throughout, and I'd appreciate if you could answer them whenever you have some time.
> Looking forward to hearing back from you!
> Cheers!
> In app/src/main/java/com/arctouch/pablo/movieapp/activity/ActMovies.java:
> > +
> +    // api
> +    public static MoviesAPI api = null;
> +
> +    // start by 1
> +    private int currentPage = 1;
> +
> +    // view components
> +    private RecyclerView listMovies;
> +    private AdtMovie adtMovie;
> +    private RecyclerView.LayoutManager mLayoutManager;
> +    private ProgressBar progressBar;
> +    private boolean isLoading = false;
> +
> +    // genres and upcoming movies
> +    private List<ResponseGenre> genres = new ArrayList<>();
> What's this list for? It doesn't seem to be used…
Really, I forgot to remove this list. I've tried to get the genres first and keep it in memory to avoid to request to each movie but in the final changes I moved to another class, and didn’t remove from it.

> In app/src/main/java/com/arctouch/pablo/movieapp/activity/ActMovies.java:
> > +    // view components
> +    private RecyclerView listMovies;
> +    private AdtMovie adtMovie;
> +    private RecyclerView.LayoutManager mLayoutManager;
> +    private ProgressBar progressBar;
> +    private boolean isLoading = false;
> +
> +    // genres and upcoming movies
> +    private List<ResponseGenre> genres = new ArrayList<>();
> +
> +    @Override
> +    protected void onCreate(Bundle savedInstanceState) {
> +        super.onCreate(savedInstanceState);
> +
> +        //
> +        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbarMovies);
> Do we really need the cast to Toolbar? Didn't findViewById get generified in sdk 26 (which we're using)?
I didn’t need, I think was routine, but you are right, I only need this cast to older sdk versions.

> In app/src/main/java/com/arctouch/pablo/movieapp/activity/ActMovies.java:
> > +        callUpcomingMovies().enqueue(new Callback<ResponseUpcomingMovies>() {
> +
> +            @Override
> +            public void onResponse(Call<ResponseUpcomingMovies> call, Response<ResponseUpcomingMovies> response) {
> +
> +                Log.d(TAG, response.message());
> +
> +                //
> +                isLoading = false;
> +
> +                // response ok
> +                if (response.isSuccessful()) {
> +
> +                    // add to adapter
> +                    adtMovie.addAll(response.body().getResults());
> +                    listMovies.setAdapter(adtMovie);
> When you change the RecyclerView's adapter, you force it to update, which causes it to scroll back to the beginning every time it loads more movies. Do you know what to change so you don't have to change the adapter on every movie update/
No, I don’t. Really bad code :/
I can remove this setAdapter and putting in the line 77, after create it.


> In app/src/main/java/com/arctouch/pablo/movieapp/adapter/AdtMovie.java:
> > +        // for each genre
> +        for (Integer genreId : movie.getGenreIds()) {
> +            genreNames += genreNames.isEmpty() ? genres.get(genreId) : ", " + genres.get(genreId);
> +        }
> +
> +        return genreNames;
> +    }
> +
> +
> +    /**
> +     * Add objects to list
> +     *
> +     * @param mcList
> +     */
> +    public void addAll(List<ResponseUpcomingMovie> mcList) {
> +        for (ResponseUpcomingMovie mc : mcList) {
> Is there a reason you didn't use movies.addAll(mcList)?
No, other bad code :/
I can use only addAll.

> In app/src/main/java/com/arctouch/pablo/movieapp/activity/ActMovies.java:
> > +    /**
> +     * Init API
> +     */
> +    private void initAPI() {
> +
> +        //
> +        Log.d(TAG, "InitAPI");
> +
> +        // build retrofit
> +        Retrofit retrofit = new Retrofit.Builder()
> +                .baseUrl("https://api.themoviedb.org/")
> +                .addConverterFactory(GsonConverterFactory.create())
> +                .build();
> +
> +        // instance api
> +        ActMovies.api = retrofit.create(MoviesAPI.class);
> If you're creating the Retrofit instance here, why do you need the RESTAdapter class?
I create RESTAdapter first, but I copied and pasted the code to made a test and forgot to remove when test was completed. I could only instance a RESTAdapter sending a URL.

> In app/src/main/java/com/arctouch/pablo/movieapp/model/ResponseUpcomingMovie.java:
> > +    }
> +
> +    /**
> +     * @return The backdropPath
> +     */
> +    public String getBackdropPath() {
> +        return backdropPath;
> +    }
> +
> +    /**
> +     * Get Genres Name From
> +     *
> +     * @param genres
> +     * @return
> +     */
> +    public String getGenresName(List<ResponseGenre> genres) {
> Isn't this the same method we have in AdtMovie? (Though this one seems incomplete…)
Yes, it is!
Wasn’t removed from this class.


> In app/src/main/res/layout/content_act_movie.xml:
> > +
> +    <TextView
> +        android:id="@+id/textViewDetailOverview"
> +        android:layout_width="0dp"
> +        android:layout_height="0dp"
> +        android:layout_marginEnd="8dp"
> +        android:layout_marginStart="8dp"
> +        android:layout_marginTop="8dp"
> +        android:text="@string/movie_overview"
> +        android:textAlignment="center"
> +        app:layout_constraintBottom_toBottomOf="parent"
> +        app:layout_constraintEnd_toEndOf="parent"
> +        app:layout_constraintStart_toStartOf="parent"
> +        app:layout_constraintTop_toBottomOf="@+id/textViewDetailReleaseDate" />
> +
> +</android.support.constraint.ConstraintLayout>
> This layout is ok, but if we turn the phone sideways (to landscape mode), then we lose half of the information... What do you think we could do avoid that problem?
We could create a landscape layout variation.

> In app/build.gradle:
> > +        }
> +    }
> +}
> +
> +dependencies {
> +    implementation fileTree(dir: 'libs', include: ['*.jar'])
> +    implementation 'com.android.support:appcompat-v7:26.1.0'
> +    implementation 'com.android.support.constraint:constraint-layout:1.0.2'
> +    implementation 'com.android.support:design:26.1.0'
> +    testImplementation 'junit:junit:4.12'
> +    androidTestImplementation 'com.android.support.test:runner:1.0.1'
> +    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.1'
> +    compile 'com.squareup.picasso:picasso:2.5.2'
> +    implementation 'com.android.support:cardview-v7:26.1.0'
> +    compile 'com.squareup.retrofit2:retrofit:2.3.0'
> +    compile 'com.squareup.retrofit2:converter-gson:2.3.0'
> Why do you have both implementation and compile dependencies here? Shouldn't they all be implementation?
Yes, should be. I copy/pasted this code. The best approach is use implementation now.
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
